artiq/firmware/runtime/mgmt.rs

```diff
@@ -84,20 +86,36 @@ fn worker(io: &Io, stream: &mut TcpStream) -> Result<(), Error<SchedError>> {
             }
             Request::ConfigWrite { ref key, ref value } => {
                 match config::write(key, value) {
-                    Ok(_)  => Reply::Success.write_to(stream),
+                    Ok(_)  => { 
+                        if key == "idle_kernel" {
+                            io.until(|| !restart_idle.get())?;
```

Comment:
This is in case the new idle kernel is overwritten with another, newer one? So in this one case the new idle kernel would not get interrupted, but otherwise generally it would?